### PR TITLE
Avoid crash for empty input string in base64 decode

### PR DESCRIPTION
**Pull request by jeffvroom**
### Intent

Addresses:  https://github.com/rstudio/rstudio-pro/issues/4757

### CODE REVIEW

File: src/cpp/shared_core/system/Crypto.cpp [hunk 1]
```diff
@@ -244,6 +244,10 @@ Error base64Encode(const unsigned char* in_data, int in_length, std::string& out
 
 Error base64Decode(const std::string& in_data, std::vector<unsigned char>& out_decoded)
 {
+   // this crashes if passed an empty string so avoid that altogether
+   if (in_data.length() == 0)
+      return Success();
```
Comment (cm421) [inline src/cpp/shared_core/system/Crypto.cpp:249]: I wasn't sure if returning success here was the right thing to do at first, but after searching around it looks like `base64Decode` of an empty string is, in fact, an empty string. (Some test examples I found: https://www.rfc-editor.org/rfc/rfc4648#section-10)

**Review by cm421 — APPROVED** [review]
LGTM

File: src/cpp/shared_core/system/Crypto.cpp (hunk 1 above)
Comment (AndrewMcClain) [reply src/cpp/shared_core/system/Crypto.cpp:249]: Thanks for this extra context!

**Review by AndrewMcClain — APPROVED** [review]
LGTM

**Pull request merged**